### DISCUSSION TITLE
Actions appear to be super slow (dock/undock for example)

### DISCUSSION THREAD

**Discussion by jessestevens5b** [iRobotEducation/create3_docs#388] [Questions]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Raspberry Pi 4 Ubuntu 22.04 server

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.1.2

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

Yes - wifi network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

Is it just me, or is running actions (undock/dock for example) super super slow?

I find it sits for quite some time (sometimes more than 10 seconds) saying "Waiting for an action server to become available..." before finally sending the goal?

Is this a problem with the iRobot Create 3 or something else? Driving the bot via teleop is immediate and has no problems so the link is functional and all topics are listed on request.

I also see when running the Turtlebot4 lite launch scripts that I get occasional prints to the terminal saying "Service start_motor unavailable" etc. Could this be related?

**Comment by alsora** [comment]
Hi,
if when you send an action goal to the robot you get the message `"Waiting for an action server to become available..."`, this indicates that the RaspberryPi is having trouble discovering the ROS nodes on the robot.

This can either indicate:
 1. a connection problem. for some reason it's taking a lot of time to send packets from the the Pi to the robot and viceversa
 2. the robot CPU is overloaded


Can you answer the following questions:
 - what happens if you send multiple goals back to back ? for example if you send this goal and as soon as it succeeds you send it again and so on. is it only the first invocation that is slow or also successive ones?
```
ros2 action send_goal /drive_distance irobot_create_msgs/action/DriveDistance "{distance: 0.5,max_translation_speed: 0.15}"
```
 - can you capture a log from the robot debug console while you send action requests ? You can find them through the webserver "logs" tab (see https://iroboteducation.github.io/create3_docs/webserver/logs/)
